tango/device_proxy.py

```diff
@@ -334,29 +339,36 @@ def __DeviceProxy__setattr(self, name, value):
 
     try:
         self.__refresh_cmd_cache()
-    except:
-        pass
+    except Exception as e:
+        if cause is None:
+            cause = e
 
     if name_l in self.__get_cmd_cache():
         raise TypeError('Cannot set the value of a command')
 
     try:
         self.__refresh_attr_cache()
-    except:
-        pass
+    except Exception as e:
+        if cause is None:
+            cause = e
 
     if name_l in self.__get_attr_cache():
         return __set_attribute_value(self, name, value)
 
     try:
         self.__refresh_pipe_cache()
-    except:
-        pass
+    except Exception as e:
+        if cause is None:
+            cause = e
 
     if name_l in self.__get_pipe_cache():
         return self.write_pipe(name, value)
 
-    return super(DeviceProxy, self).__setattr__(name, value)
+    try:
+        return super(DeviceProxy, self).__setattr__(name, value)
+    except Exception as e:
+        raise e from cause
+
```

Comment:
We don't need this blank line.  Just 2 between functions according to PEP8.  I'm not sure why Codacy didn't complain.